### PR TITLE
Custom detail layout

### PR DESCRIPTION
**Pull request by nielsvanvelzen**
Unfinished but I can already share the first screenshots!

image
image

### CODE REVIEW

**Comment by nielsvanvelzen** [comment]
image

New screenshot (not all changes are pushed yet)

**Comment by nielsvanvelzen** [comment]
image

File: app/src/main/java/org/jellyfin/androidtv/details/DetailsDescriptionPresenter.kt [hunk 1]
```diff
@@ -1,16 +1,156 @@
...
+				viewHolder.textStreamValue.visibility = View.VISIBLE
+			} else {
+				viewHolder.textStreamLabel.visibility = View.GONE
+				viewHolder.textStreamValue.visibility = View.GONE
+			}
+
+			viewHolder.streams.visibility = View.VISIBLE
+
+			viewHolder.duration.text = TimeUtils.formatMillis(item.durationTicks / 10000)
+			viewHolder.endsAt.text = DateFormat.getTimeFormat(viewHolder.view.context).format(System.currentTimeMillis() + (item.durationTicks - item.playbackPositionTicks) / 1000)
```
Comment (AndreasGB) [inline app/src/main/java/org/jellyfin/androidtv/details/DetailsDescriptionPresenter.kt]: As mentioned in chat, should be 10000

File: app/src/main/res/layout/row_details_description_genre_button.xml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Button xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    style="@style/Button.Outline"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:minHeight="0dp"
+    android:minWidth="0dp"
+    tools:text="Fantasy"
```
Comment (nielsvanvelzen) [inline app/src/main/res/layout/row_details_description_genre_button.xml]: Wait.. when did I do this lol

File: app/src/main/java/org/jellyfin/androidtv/ui/widget/Rating.kt [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+import android.widget.TextView
+import org.jellyfin.androidtv.R
+import org.jellyfin.androidtv.util.dp
+import kotlin.math.roundToInt
+import kotlin.properties.Delegates
+
+class Rating(context: Context, attrs: AttributeSet?) : RelativeLayout(context, attrs) {
+	private val attributes = context.theme.obtainStyledAttributes(attrs, R.styleable.Rating, 0, 0)
+
+	private val imageView = ImageView(context).apply {
```
Comment (AndreasGB) [inline app/src/main/java/org/jellyfin/androidtv/ui/widget/Rating.kt]: Why do you create the elements manually and not using a layout file?

Comment (nielsvanvelzen) [reply app/src/main/java/org/jellyfin/androidtv/ui/widget/Rating.kt]: It's simple enough to not use a XML file. I hate XML files. If I don't need to use them I don't.

**Comment by NebelNidas** [comment]
The poster looks a bit small to me, especially when compared to jellyfin-web. Could its size be bumped up a little? Here are a few sizes I tested:
Big Posters - Top

**Pull request merged**